pkg/controllers/gateway/gateway_controller.go

```diff
@@ -117,9 +117,32 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
+	// TODO: Use return value
+	params, err := getParams(ctx, r.Client, gatewayClass)
+	if err != nil && !IsInvalidParamsError(err) {
+		return ctrl.Result{}, nil
+	}
+
+	// If the GatewayClass parametrers are invalid, update the status
+	// and stop reconciling
+	if IsInvalidParamsError(err) {
```

Comment:
Nit: This might read better as 

``` 
if err !=nil && IsInvalidParamsError(err) {
...

}

if err != nil{
   return ctrl.Result{}, nil
}
```

Why do we ignore the err on line 122 should that be returned?

Reply:
Makes sense, I'll make those changes. About the err being ignored, I missed it, it should indeed be returned